plugins/out_kinesis_firehose/firehose_api.c

```diff
@@ -216,8 +217,8 @@ static int process_event(struct flb_firehose *ctx, struct flush *buf,
                          ctx->delivery_stream);
             return 2;
         }
-        /* guess space needed to write time_key */
-        len = 6 + strlen(ctx->time_key) + 6 * strlen(ctx->time_key_format);
+        /* format time output and get the length */
+        len = create_time_format_string(&out_buff, ctx->time_key_format, tms);
         /* how much space do we have left */
         tmp_size = (buf->tmp_buf_size - buf->tmp_buf_offset) - written;
         if (len > tmp_size) {
```

Comment:
you need to free out_buff here

Reply:
free below

Reply:
No. If you call return, free will not be called